contracts/BalanceScanner.sol

```diff
@@ -57,6 +58,69 @@ contract BalanceScanner {
     }
   }
 
+  /**
+   * @notice Get the ERC721 owner addresses for multiple tokenIds from a single contractAddress
+   * @param token The address of the contract
+   * @param tokenIds The tokenIds we want to fetch owner for
+   * @return results The list of owner addresses of each tokenId
+   */
+
+  function ownersOfTokenIdsForToken(address token, uint256[] calldata tokenIds)
+    external
+    view
+    returns (Result[] memory results)
+  {
+    results = new Result[](tokenIds.length);
+
+    for (uint256 i = 0; i < tokenIds.length; i++) {
+      bytes memory data = abi.encodeWithSignature("ownerOf(uint256)", tokenIds[i]);
+      results[i] = staticCall(token, data, 20000);
+    }
+  }
+
+  /**
+   * @notice Get the ERC721 owner addresses for multiple tokenIds from multiple contractAddresses
+   * @param contracts The array of contract addresses to fetch owners of tokenId from
+   * @param tokenIds The array of tokenIds to fetch owners for
+   * @return results The list of owner addresses of each tokenId and contractAddress
+   */
+
+  function ownersOfTokenIdsForTokens(address[] calldata contracts, uint256[] calldata tokenIds)
+    external
+    view
+    returns (Result[] memory results)
+  {
+    require(contracts.length == tokenIds.length, "Contracts and tokenIds array length mismatch");
```

Comment:
If `uint256[][]` is a valid function argument we can instead do 
```
  function ownersOfTokenIdsForTokens(address[] calldata contracts, uint256[][] calldata tokenIds)
```

instead of duplicate `address` for mapping from `address  => tokenId`

For example:
Looking up owner of token ids `[1,2,3]` on collection `0x0001000200010002000100020001000200010002` and `[1,2]` on collection `0x0001000100010001000100010001000100010001`
can be 
```
ownersOfTokenIdsForTokens(
   ['0x0001000200010002000100020001000200010002', '0x0001000100010001000100010001000100010001'],
   [[1,2,3],[[1,2]]]
)
```
instead of 
```
ownersOfTokenIdsForTokens(
   ['0x0001000200010002000100020001000200010002', '0x0001000200010002000100020001000200010002', '0x0001000200010002000100020001000200010002', '0x0001000100010001000100010001000100010001','0x0001000100010001000100010001000100010001'],
   [1,2,3,1,2]
)
```

Reply:
I have pushed a change for that, basically, we are making use of `ownersOfTokenIdsForToken ` that takes a  contract address and array of tokenIds;